vortex-cxx/cpp/include/vortex/expr.hpp

```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: Copyright the Vortex contributors
+
+#pragma once
+
+#include "vortex/scalar.hpp"
+#include "vortex_cxx_bridge/lib.h"
+#include <vector>
+
+namespace vortex {
+
+class Expr {
+public:
+    Expr() = delete;
+    Expr(Expr &&other) noexcept = default;
+    Expr &operator=(Expr &&other) noexcept = default;
+    ~Expr() = default;
+
+    Expr(const Expr &) = delete;
+    Expr &operator=(const Expr &) = delete;
+
+    static Expr literal(Scalar scalar);
```

Comment:
I think all static functions introduced as part of this PR should become free standing functions, living in a nested a namespace within vortex. 

We can define the nested namespace in one go:
```
namespace vortex::expr {
    //...
}
```

Reply:
Thanks, fixed